File: nodecore-spv/src/main/java/veriblock/model/FutureEventReply.java
package veriblock.model;

import nodecore.api.grpc.VeriBlockMessages;

import java.util.concurrent.atomic.AtomicBoolean;

public class FutureEventReply {

    private final AtomicBoolean isDone = new AtomicBoolean(false);
    private VeriBlockMessages.Event response;

    public synchronized void response(VeriBlockMessages.Event event) {
        this.response = event;
        this.isDone.set(true);
    }

    public boolean isDone() {
        return isDone.get();
    }

    public VeriBlockMessages.Event getResponse() {
        return response;
    }
}

File: nodecore-spv/src/main/java/veriblock/model/FutureEventReply.kt
package veriblock.model

import nodecore.api.grpc.VeriBlockMessages
import java.util.concurrent.atomic.AtomicBoolean

class FutureEventReply {
    private val isDone = AtomicBoolean(false)

    lateinit var response: VeriBlockMessages.Event
        private set

    @Synchronized
    fun response(event: VeriBlockMessages.Event) {
        response = event
        isDone.set(true)
    }

    fun isDone(): Boolean {
        return isDone.get()
    }
}
